dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt

```diff
@@ -25,62 +27,102 @@ import java.util.*
-        /** @see PluginFeaturesService.v2PluginEnabled */
-        val v2PluginEnabled: Property<Boolean>
-
-        /** @see [PluginFeaturesService.v2PluginNoWarn] */
-        val v2PluginNoWarn: Property<Boolean>
-
-        /** @see [PluginFeaturesService.v2PluginMigrationHelpersEnabled] */
-        val v2PluginMigrationHelpersEnabled: Property<Boolean>
-
         /** @see [PluginFeaturesService.primaryService] */
         val primaryService: Property<Boolean>
 
+        /** @see PluginFeaturesService.pluginMode */
+        val pluginMode: Property<String>
+
+        /** If `true`, suppress [pluginMode] messages. */
+        val pluginModeNoWarn: Property<Boolean>
+
         /** If `true`, enable K2 analysis. */
         val k2AnalysisEnabled: Property<Boolean>
 
-        /** If `true`, suppress the K2 analysis message. */
+        /** If `true`, suppress [k2AnalysisEnabled] messages. */
         val k2AnalysisNoWarn: Property<Boolean>
+
+        /** [Project.getDisplayName] - only used for log messages. */
+        val projectDisplayName: Property<String>
+
+        /** [Project.getProjectDir] - only used for log messages. */
+        val projectDirectory: Property<String>
     }
 
     /**
      * Designate this [BuildService] as 'primary', meaning it should log messages to users.
      * Non-primary services should not log messages.
      *
-     * Why? Because Gradle is buggy. Sometimes registering a BuildService fails.
-     * See https://github.com/gradle/gradle/issues/17559.
-     * If service registration fails then re-register the service, but with a distinct name
-     * (so it doesn't clash with the existing but inaccessible BuildService), and don't mark it as 'primary'.
+     * Why? Two reasons:
+     *
+     * 1. Sometimes registering a BuildService fails.
+     *   See https://github.com/gradle/gradle/issues/17559.
+     *   If service registration fails then re-register the service, but with a distinct name
+     *   (so it doesn't clash with the existing but inaccessible BuildService),
+     *   and don't mark it as 'primary' to avoid duplicate logging.
+     * 2. The method Gradle uses to generates accessors for pre-compiled script plugins
+     *   (see [PluginFeaturesService.Companion.configureParamsDuringAccessorsGeneration])
+     *   runs in a temporary project that is evaluated twice.
      *
      * @see org.jetbrains.dokka.gradle.internal.registerIfAbsent
      */
-    private val primaryService: Boolean get() = parameters.primaryService.getOrElse(false)
+    private val primaryService: Boolean
+        get() = parameters.primaryService.getOrElse(false)
 
     /**
      * Whether DGP should use V2 [org.jetbrains.dokka.gradle.DokkaBasePlugin].
      *
-     * Otherwise, fallback to V1 [org.jetbrains.dokka.gradle.DokkaClassicPlugin].
+     * @see pluginMode
      */
     internal val v2PluginEnabled: Boolean by lazy {
-        val v2PluginEnabled = parameters.v2PluginEnabled.getOrElse(false)
+        when (pluginMode) {
```

Comment:
minor: would be nice to have `v2PluginMigrationHelpersEnabled` flag located just below `v2PluginEnabled` for easier understanding during reading of the code

Reply:
I agree, done.